server/util/flagutil/flagutil.go

```diff
@@ -275,16 +182,42 @@ func (f *URLFlag) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return err
 }
 
+func (f *URLFlag) AliasedType() reflect.Type {
```

Comment:
do we need both AliasedType and YAMLTypeAlias()? the url flag is defined as a string so i think if you were to return *URLFlag in AliasedType it would still work. in what cases would they be different?

Reply:
We do; I use it in https://github.com/buildbuddy-io/buildbuddy/pull/2014, for example. The `AliasedType` is the "native" type of the flag in go (the type returned to the user when creating the flag), while the `YAMLTypeAlias` is the type used by YAML, which will importantly have the `MarshalYAML` and `UnmarshalYAML` methods.